Add first version of amtrak script

sverch opened the pull request:
This has two modes:

- Read specific trips including train numbers from a CSV to add them up
- Dump all results for a single search query

See README for usage.

Commit:
Add first version of amtrak script

This has two modes:

- Read specific trips including train numbers from a CSV to add them up
- Dump all results for a single search query

See README for usage.

SYU15 commented:
Nice!

jisaacso commented on amtrakomatic/do_all_searches.sh:
```diff
@@ -0,0 +1,51 @@
+
+
+
+#run python main.py --source NewYork --destination Boston --date  08/11/2019
```
all the comments

jisaacso commented on amtrakomatic/extract_expected_result.sh:
```diff
@@ -0,0 +1,17 @@
+    cat <<EOF
+Usage: $0
+
+    Runs pytest and vomits the json of the expected output for whatever failed.
```
I would suggest a real test runner, e.g., tox

jisaacso commented on amtrakomatic/fixup.sh:
```diff
@@ -0,0 +1,13 @@
+echo '[]' >| test_data/seattle_chicago_08_24_2019_False_0.json 
```
why?

jisaacso commented on amtrakomatic/format_output.py:
```diff
@@ -0,0 +1,57 @@
+"""
+Library to nicely format the result output.
+"""
+def print_result(result):
```
more descriptive name?

jisaacso commented on amtrakomatic/fuzzy_match.py:
```diff
@@ -0,0 +1,42 @@
+    Given a rough station name, does a fuzzy match on all Amtrak stations to return the proper
+    station code.
+    """
+    station_map = load_stations()
```
calls load for every search?

jisaacso reviewed (changes requested):
Only you would go so far _off the rails_ . Overall looks good.

Commit:
Get back on track

- Used tox for testing
- Remove commented out code
- Remove temporary test fixup scripts
- Split up directories
- Only load station list once
- Move printer into AmtrakResults class for clarity

Commit:
Unload some unnecessary cargo

This script was a test helper that doesn't even work anymore.

sverch commented:
I've trained my whole life for this moment @jisaacso...
